src/vector/geometry.rs

```diff
@@ -233,9 +233,19 @@ impl Geometry {
         Ok(())
     }
 
+    /// Compute geometry area in units of the spatial reference system in use.
+    ///
+    /// Supported for `Curve` (including `LineString` and `CircularString`) and `MultiCurve`.
+    /// Returns zero for all other geometry types.
+    ///
+    /// See: [`OGR_G_Length`](https://gdal.org/api/vector_c_api.html#_CPPv412OGR_G_Length12OGRGeometryH)
+    pub fn length(&self) -> f64 {
+        unsafe { gdal_sys::OGR_G_Length(self.c_geometry()) }
+    }
+
     /// Compute geometry area in square units of the spatial reference system in use.
     ///
-    /// Supported for `LinearRing`, `Polygon` or `MultiPolygon`.
+    /// Supported for `LinearRing`, `Polygon` and `MultiPolygon`.
```

Comment:
Drive-by nit: I know it came from the GDAL docs, but I think this reads better than "or".